src/distributions/integer.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// https://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The implementations of the `Uniform` distribution for integer types.
+
+use core::mem;
+
+use {Rng};
+use distributions::{Distribution, Uniform};
+
+impl Distribution<isize> for Uniform {
+    #[inline]
+    fn sample<R: Rng>(&self, rng: &mut R) -> isize {
+        if mem::size_of::<isize>() == 4 {
+            rng.gen::<i32>() as isize
+        } else {
+            rng.gen::<i64>() as isize
```

Comment:
What about platforms with 16 bit pointers? Does Rust support them?

Reply:
I don't think so. There was some discussion of 16-bit CPUs [here](https://github.com/dhardy/rand/issues/11) but ultimately because of the extremely limited memory addressing I think most 16-bit code has to be written expressly for that architecture.